airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -32,14 +34,42 @@
     dataset_schema,
 )
 from airflow.models.dataset import DatasetEvent, DatasetModel
+from airflow.security import permissions
 from airflow.utils.db import get_query_count
+from airflow.utils.log.action_logger import action_event_from_permission
 from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.www.decorators import action_logging
 
 if TYPE_CHECKING:
     from sqlalchemy.orm import Session
 
     from airflow.api_connexion.types import APIResponse
 
+RESOURCE_EVENT_PREFIX = "dataset"
+
+
+@security.requires_access_dataset("DELETE")
+@provide_session
+@action_logging(
+    event=action_event_from_permission(
+        prefix=RESOURCE_EVENT_PREFIX,
+        permission=permissions.ACTION_CAN_DELETE,
+    ),
+)
+def delete_dataset(*, uri: str, session: Session = NEW_SESSION) -> APIResponse:
+    """Delete a Dataset."""
+    dataset = session.scalar(select(DatasetModel).where(DatasetModel.uri == uri))
+    if dataset is None:
+        raise NotFound(
+            "Dataset not found",
+            detail=f"The Dataset with uri: `{uri}` was not found",
+        )
+    if dataset.consuming_dags or dataset.producing_tasks:
+        raise AlreadyExists(detail="Dataset is still referenced by other DAG")
```

Comment:
It doesn't seem like this is a good use case for "AlreadyExists", maybe we should create a new custom exception

Reply:
@edumuellerFSL what about the HTTP Status Code? Since 409 is already mapped to `AlreadyExists` do you have a suggestion on what should the status code be for this case?

Reply:
imo using status code 409 is correct here

Reply:
@edumuellerFSL do you think we should generalize the `AlreadyExists` error (e.g change to `ResourceConflict`) or create a new custom exception but use the same 409 code as `AlreadyExists` ? 

Because from the existing [EXCEPTIONS_LINK_MAP](https://github.com/apache/airflow/blob/f3cc5f505bf6dad696d3cb496a77bc3a0f5bf45d/airflow/api_connexion/exceptions.py#L32C1-L32C20) structure I'm not sure if I can add another mapping with same key.

Reply:
You're right, it won't be possible to add another mapping with the same key. I think generalizing it is a good alternative.